### PR TITLE
chore: Added .NET 6 support, and removed .NET Core 2 and 3 support

### PR DESCRIPTION
**Pull request by LeftTwixWand**
I propose to remove .NET Core 2 and 3 support, because officially those versions already out of support: [official site](https://dotnet.microsoft.com/download/dotnet)
What about .NET Core 2.1 - the end of support will be at this August.
And .NET Core 3.1 will be alive 6 months more.

So I propose to update the packing tool and start using .NET 5 and .NET 6 only.

### CODE REVIEW

File: Packaging.Targets.Tests/Packaging.Targets.Tests.csproj [hunk 1]
```diff
@@ -78,5 +78,8 @@
   <ItemGroup>
     <AdditionalFiles Include="..\Packaging.Targets\stylecop.json" />
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Update="NerdBank.GitVersioning" Version="3.4.231" />
```
Comment (qmfrederik) [inline Packaging.Targets.Tests/Packaging.Targets.Tests.csproj:82]: Can you update this in `MSBuild.Directory.props` instead?

File: dotnet-deb/dotnet-deb.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp2.1;netcoreapp3.0;net5.0</TargetFrameworks>
+    <TargetFrameworks>net5.0;net6.0</TargetFrameworks>
```
Comment (qmfrederik) [inline dotnet-deb/dotnet-deb.csproj:4]: Let's keep .NET Core 2.1 and 3.1 until they are end of life.

File: dotnet-rpm/PackagingRunner.cs [hunk 3]
```diff
@@ -44,7 +43,7 @@ public PackagingRunner(string outputName, string msbuildTarget, string commandNa
 
         public int Run(string[] args)
         {
-            CommandLineApplication commandLineApplication = new CommandLineApplication(throwOnUnexpectedArg: true);
+            CommandLineApplication commandLineApplication = new CommandLineApplication();
```
Comment (qmfrederik) [inline dotnet-rpm/PackagingRunner.cs:46]: Why is this change needed in the context of this PR?

**Review by qmfrederik — CHANGES_REQUESTED** [review]
Thanks, adding `net6.0` support looks good to me. I'm not so sure about dropping `netcoreapp2.1` and `netcoreapp3.x` support while they are still under support; let's keep them until they are formally out of support.

File: dotnet-deb/dotnet-deb.csproj (hunk 2 above)
Comment (LeftTwixWand) [reply dotnet-deb/dotnet-deb.csproj:4]: The problem is, if we want to save .NET Core 2.1 and use .NET 6 - we have to use to many preprocessor directives, like
`#if NET5_0_OR_GREATER`

File: dotnet-rpm/PackagingRunner.cs (hunk 3 above)
Comment (qmfrederik) [reply dotnet-rpm/PackagingRunner.cs:46]: For `McMaster.Extensions.CommandLineUtils`, the ambition is to replace that dependency by [`System.CommandLine`](https://github.com/dotnet/command-line-api). I'd rather not update it if it's not required (e.g. to fix a bug); I've been bitten by breaking changes in the command line parsing packages in the past :-)

File: dotnet-deb/dotnet-deb.csproj (hunk 2 above)
Comment (LeftTwixWand) [reply dotnet-deb/dotnet-deb.csproj:4]: Something like this:
image

As I understood by description - Microsoft.Build 16.10 can work only with .NET Framework 4.7.2 or .NET 5 or higher
image

Comment (qmfrederik) [reply dotnet-deb/dotnet-deb.csproj:4]: Thanks for clarifying! In that case, I'd _really_ appreciate it if we can defer updating those dependencies a bit, and have this PR focus on updating the target frameworks.

I'm fine with updating dependencies which don't influence the list of supported .NET Core versions (e.g. the test SDKs or NerdBank.GitVersioning).

I'd be happy to drop .NET Core 2 support end of August, but dropping .NET Core 3 may need a bit more of convincing ;-).

Comment (LeftTwixWand) [reply dotnet-deb/dotnet-deb.csproj:4]: Ok, I will check all the references, and try to configure it for the .NET Core 3.1 and higher versions :-)

File: dotnet-rpm/PackagingRunner.cs (hunk 3 above)
Comment (ahopper) [reply dotnet-rpm/PackagingRunner.cs:46]: Is this simpler now >netCore 2.1 is eol ?

**Comment by keesschollaart81** [comment]
Would love this PR to be merged. It's one of my last dependencies that's blocking me from upgrading to NET6 today. Dropping support for net3.1 is not a huge issue, right? For the remaining 6 months, users can continue to use the current 'stable' 0.1.189 version which has been there for a year already...?!

**Comment by qmfrederik** [comment]
I'm working on a couple of PRs which should help prepare for .NET 6 support, so hopefully we can get that soon (via #198).
This PR does two things: update all dependencies + target .NET 6, which made it a bit harder to digest.

**Comment by qmfrederik** [comment]
Version 0.1.217 of the dotnet-packaging tools support `net6.0`. Please give it a try!

**Comment by qmfrederik** [comment]
@LeftTwixWand With net6.0 now being in master, can we go ahead and close this PR? Should you want to work on updating the dependencies of this project, feel free to do so as well. `netcoreapp3.1` is supported until December 2022, and I would prefer to keep support for `netcoreap3.1` until then.

**Comment by qmfrederik** [comment]
@LeftTwixWand I'll go ahead and close this PR, as net 6.0 support has landed. Thanks!
